JWBestPracticeApps/DRM Fairplay/DRM Fairplay/ViewController.swift

```diff
@@ -8,38 +8,63 @@
 import UIKit
 import JWPlayerKit
 
-private let EZDRMLicenseAPIEndpoint = "https://fps.ezdrm.com/api/licenses"
-private let EZDRMCertificateEndpoint = "https://fps.ezdrm.com/demo/video/eleisure.cer"
-private let EZDRMVideoEndpoint = "https://fps.ezdrm.com/demo/video/ezdrm.m3u8"
-
 class ViewController: JWPlayerViewController,
                       JWDRMContentKeyDataSource {
-
-    var contentUUID: String?
-
+    
+    /// All inputs required for the FairPlay DRM process (the "App" in figure 1-2 of [FairPlay Streaming Overview.pdf](https://developer.apple.com/streaming/fps/FairPlayStreamingOverview.pdf),
+    /// page 6) are prefetched and prepared ahead of time, and briefly stored in this struct for use in the content
+    /// key protocol's handlers.
+    var drmProperties: DRMContentKeyRequestInputs?
+
+    /// Composed of the media ID, DRM policy ID, and a JWT generated from the two IDs + the property's client secret.
+    var signedVideoURL: URL {
+        guard var mutableUrl = URL(string: "https://cdn.jwplayer.com/v2/")
+        else { fatalError("App demo failed on: line \(#line)") }
+        
+        mutableUrl.append(path: "media/\(JWTestAsset.mediaId)")
+        mutableUrl.append(path: "drm/\(JWTestAsset.policyId)")
+        
+        // FIXME: This can and should be replaced with your preferred JWT generation method.
+        let token = TokenGenerator.createToken(
```

Comment:
I think this is missing the signature, which is the API secret and the essential reason why we shouldn't encourage generating signed URLs from an app. 
https://docs.jwplayer.com/players/docs/ios-apply-studio-drm-with-jw-platform#implementation 
> We strongly recommend using a proxy service to generate the JSON web token (JWT). If you generate the JWT within a client-side native app, you risk exposing your API secret.

Documentation for JWT signed URLs
https://docs.jwplayer.com/platform/reference/protect-your-content-with-signed-urls#generate-the-jwt-signed-url